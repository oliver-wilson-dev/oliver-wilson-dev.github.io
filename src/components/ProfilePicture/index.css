.profilePicture {
    vertical-align: middle;
    border-radius: 50%;
    box-shadow: 3px 3px 10px #0b0a0a;
    object-fit: cover;
    width: 200px;
    height: 200px;
    filter: grayscale(100%);
    margin: 20px auto 0px auto
}