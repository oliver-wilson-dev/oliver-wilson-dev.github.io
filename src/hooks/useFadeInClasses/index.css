@import "../../styles/transitions.css";

.hiddenFadeIn {
    transition: visibility var(--sectionVisibilityTransitionDuration), opacity var(--sectionOpacityTransitionDuration);
    transition-delay: var(--spinnerOpacityTransitionDuration);
    opacity: 0;
    visibility: hidden;

    &.show {
        opacity: 1;
        visibility: visible;
    }
}