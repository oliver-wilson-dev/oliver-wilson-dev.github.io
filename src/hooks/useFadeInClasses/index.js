import { useEffect, useState } from 'react';
import sectionStyles from './index.css';

const useFadeInClasses = () => {
  const [visible, setVisible] = useState(false);

  useEffect(() => {
    setVisible(true);
  }, []);

  const fadeInClasses = {
    [sectionStyles.hiddenFadeIn]: true,
    [sectionStyles.show]: visible
  };

  return { fadeInClasses };
};

export default useFadeInClasses;
