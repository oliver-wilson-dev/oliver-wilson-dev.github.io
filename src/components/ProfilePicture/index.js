import React from 'react';
import classnames from 'classnames';
import styles from './index.css';

import useFadeInClasses from '../../hooks/useFadeInClasses';


const ProfilePicture = () => {
  const { fadeInClasses } = useFadeInClasses();

  return (
    <img
      className={classnames(styles.profilePicture, fadeInClasses)}
      src="public/img/profile-pic.jpg"
      alt="A shot of me"
    />
  );
};

export default ProfilePicture;
