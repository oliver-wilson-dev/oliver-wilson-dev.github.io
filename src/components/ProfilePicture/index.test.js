import React from 'react';
import { shallow } from 'enzyme';
import ProfilePicture from '.';
import useFadeInClasses from '../../hooks/useFadeInClasses';

jest.mock('../../hooks/useFadeInClasses');


describe('ProfilePicture', () => {
  beforeEach(() => {
    useFadeInClasses.mockReturnValueOnce({ fadeInClasses: 'test-fade-in-classes' });
  });


  const defaultProps = {};

  const render = ({ overrideProps } = { overrideProps: {} }) => shallow(
    <ProfilePicture
      {...defaultProps}
      {...overrideProps}
    />
  );

  it('should render correctly', () => {
    expect(render()).toMatchSnapshot();
  });
});
