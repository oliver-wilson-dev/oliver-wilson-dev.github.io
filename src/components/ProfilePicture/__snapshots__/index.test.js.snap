// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePicture should render correctly 1`] = `
<img
  alt="A shot of me"
  className="profilePicture test-fade-in-classes"
  height="200"
  src="public/img/profile-pic.jpg"
  width="200"
/>
`;
